yarn-project/simulator/src/public/executor.ts

```diff
@@ -113,18 +104,6 @@ export class PublicExecutor {
       // (which counts the request itself)
     );
 
-    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/5818): is this really needed?
```

Comment:
These checks are also repeated on the private side so illegal calls are caught as soon as possible (otherwise we would have to wait until the kernel checks). I'm not 100% sure on how this is handled on the avm sim side, but might be worth looking at the tradeoff in speed (static calls e2e before/after should provide the relevant data)

Reply:
This check was happening after AVM simulation, but it is the job of the AVM to prevent/catch this. If this check ever failed it would mean that the AVM is broken.

Tests:
* https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/simulator/src/avm/avm_simulator.test.ts#L853
* https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/simulator/src/avm/opcodes/accrued_substate.test.ts#L368